torchquad/utils/set_precision.py

```diff
@@ -0,0 +1,30 @@
+import torch
+import os
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def set_precision(data_type="float"):
+    """This function allows the user to set the default precision. Call before declaring any variables!
+    NB: Remember to initialize CUDA first if GPU compatibility is desired.
+
+    Args:
+        data_type (string, optional): Data type to use, either "float" or "double". Defaults to "float".
+    """
+
+    if torch.cuda.is_initialized():
+        if data_type.lower() == "double":
+            print("CUDA is initialized. Setting default tensor type to cuda.Float64")
```

Comment:
Probably we should start using logging instead of print, I think.  (Then you can also keep the logger!) (It is buggy in notebooks at the moment though, have created a ticket for that)

Reply:
I'd log this with logger.warn("") probably just to make sure it is displayed by default? (If people want high-accuracy integrals accidentally switching everything to float may be very inconvenient)